### PR TITLE
Sub app bar

### PR DESCRIPTION
**Pull request by mirimkim00**

### CODE REVIEW

File: lib/components/sub_pages/program_details.dart [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+import 'package:sport_manager/services/convert_timestamp.dart';
+import '../../../services/firebase_service.dart';
+import '../../../assets/constants.dart' as constants;
+import 'package:sport_manager/components/home_page/white_container.dart';
+import 'package:sport_manager/components/sub_pages/sub_appBar.dart';
+import 'package:sport_manager/components/home_page/program_page/program_main.dart';
 
 class ProgramDetails extends StatelessWidget {
   const ProgramDetails({super.key});
+  static const IconData home_filled = IconData(0xe319, fontFamily: 'MaterialIcons');
```
Comment (maciejdrobnik) [inline lib/components/sub_pages/program_details.dart:14]: What is 0xe319? I mean it is probably good value but it is probably hard to say what it is in a month.

File: lib/components/sub_pages/program_details.dart [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
 
   @override
   Widget build(BuildContext context) {
-    return Container();
+    FirebaseService firebaseService = FirebaseService();
+    ConvertTimestampService timestampService = ConvertTimestampService();
+
+    return Scaffold(
+      appBar: subAppBar(context, titleText: 'Match information'),
+      backgroundColor: const Color(constants.homePageBackground),
```
Comment (maciejdrobnik) [inline lib/components/sub_pages/program_details.dart:23]: Nice that you used the constants but the name is not really related with the usage. You are setting the background of the appbar not background of homePage

File: lib/components/sub_pages/sub_appBar.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import '../../../assets/constants.dart' as constants;
+
+AppBar subAppBar(BuildContext context, {String? titleText}){
```
Comment (maciejdrobnik) [inline lib/components/sub_pages/sub_appBar.dart:4]: The name `subAppBar `is not meaningfull enough for me. I would name it for example `SubPagesAppBar`.
Another thing in naming is that in dart the name of the class should be written in UpperCamelCase.
You can take a look here: https://dart.dev/guides/language/effective-dart/style

File: lib/components/sub_pages/sub_appBar.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+              width: 40,
+              child: IconButton(
+              icon: const Icon(Icons.arrow_back_ios_rounded),
+              onPressed: () {
+                Navigator.pop(context);
+              },
+              ),
+            ),
+            SizedBox(
+              width: 40,
```
Comment (maciejdrobnik) [inline lib/components/sub_pages/sub_appBar.dart:21]: Last magic number

File: lib/components/sub_pages/sub_appBar.dart [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+              child: IconButton(
+              icon: const Icon(homeFilled),
+              onPressed: () {
+                Navigator.popUntil(context, ModalRoute.withName('/'));
```
Comment (maciejdrobnik) [inline lib/components/sub_pages/sub_appBar.dart:25]: Why popUntil? Single pop wouldn't be enough?

File: lib/components/sub_pages/program_details.dart (hunk 2 above)
Comment (mirimkim00) [reply lib/components/sub_pages/program_details.dart:23]: I'm afraid I can't get your point. The appbar has been done in subAppBar widget.

**Review by maciejdrobnik — APPROVED** [review]
Looks good

**Pull request merged**